### PR TITLE
Add value class run tests

### PR DESCRIPTION
**Pull request by smarter**
Review by @DarkDimius .

### CODE REVIEW

File: tests/pending/run/t6260-delambdafy.check [hunk 1]
```diff
@@ -1,4 +0,0 @@
-f(C@2e)
```
Comment (DarkDimius) [inline tests/pending/run/t6260-delambdafy.check]: Why were those files deleted?

File: src/dotty/tools/dotc/transform/ExtensionMethods.scala [hunk 2]
```diff
@@ -88,11 +88,10 @@ class ExtensionMethods extends MiniPhaseTransform with DenotTransformer with Ful
             decls1.enter(evt2uSym)
           }
 
-          // add a VCXXXCompanion superclass
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/transform/ExtensionMethods.scala]: I'd better leave this comment as is, it explains what should be happening in next lines.

File: tests/run/t6260-delambdafy.scala [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+object Test {
+  val f = (x: C[Any]) => {println(s"f($x)"); x}
+  def main(args: Array[String]): Unit = {
+    f(new C("."))
+
+    // Lambdas in Dotty are created with LambdaMetaFactory (when possible)
+    // which does not guarantee that the list of methods will always be the same.
+    /*
+    val methods = f.getClass.getDeclaredMethods.map(_.getName).sorted
```
Comment (DarkDimius) [inline tests/run/t6260-delambdafy.scala]: This test tests what happens in this particular place: that lambda is created. It wouldn't be created if method required an outer-pointer. It should continue to list methods, but check file should not contain `Test$lambda$1$$apply` anymore.

File: tests/run/value-class-extractor.scala [hunk 4]
```diff
@@ -6,12 +6,12 @@ object NonNullChar {
...
 }
 
-final class SomeProduct extends Product3[String, Int, List[String]] {
+final class SomeProduct /*extends Product3[String, Int, List[String]]*/ {
   def canEqual(x: Any) = x.isInstanceOf[SomeProduct]
   def _1 = "abc"
   def _2 = 5
   def _3 = List("bippy")
-  def isEmpty = false
+  def isDefined = true
```
Comment (DarkDimius) [inline tests/run/value-class-extractor.scala]: We want to try maintaining tests compatible with `scalac` as much as possible. It's better to maintain `isEmpty` method.

File: tests/pending/run/t6260-delambdafy.check (hunk 1 above)
Comment (smarter) [inline tests/pending/run/t6260-delambdafy.check]: They were moved, not deleted, but the files changed too much for git to detect the move.

**Comment by smarter** [comment]
Addressed all comments.

**Comment by smarter** [comment]
@DarkDimius: OK, added back the printing of the methods.
